.github/workflows/build.yml

```
@@ -0,0 +1,94 @@
+  build:
+    # Skip auto-build branches and fork PRs (no push access).
+    if: >-
+      !startsWith(github.head_ref, 'auto-build/')
+      && !startsWith(github.ref_name, 'auto-build/')
+      && (github.event_name != 'pull_request'
+          || github.event.pull_request.head.repo.full_name == github.repository)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Determine target branch
+        id: target
+        run: |
```

actionlint (1.7.12)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)